rdt/transformers/categorical.py

```diff
@@ -40,7 +40,7 @@ class UniformEncoder(BaseTransformer):
     """
 
     INPUT_SDTYPE = 'categorical'
-    SUPPORTED_SDTYPES = ['categorical', 'boolean']
+    SUPPORTED_SDTYPES = ['categorical', 'boolean', 'id', 'text']
```

Comment:
SDV currently assigns columns with the sdtype `id` to `text`, so I included both as supported.

Reply:
should we update sdv?

Reply:
We can, I can file an issue for it. Should we remove text here in that case?

Reply:
Maybe we can ask @npatki ? I think it's probably fine to leave text

Reply:
I remember `text` was the old name of the sdtype that eventually became `id`. Since we updated RDT library first to 1.0, we had called it `text`. But ultimately in SDV 1.0, we decided to call it `id`. 

I thought we went back to RDT and renamed `text` to `id` everywhere. Is that not the case?

Reply:
We did rename `'text'` to `'id'`, but left `'text'` as a supported sdtype for all the ID transformers. The SDV data processor [still assigns the `'text'` sdtype to ID columns](https://github.com/sdv-dev/SDV/blob/391fd824fad082a97207705c522a61b7ae0ca68a/sdv/data_processing/data_processor.py#L574) when it creates the config for the hyper transformer.

Reply:
The journey we want to support it:
1. Metadata has listed a column's sdtype as `id` without any `regex_format`; this column is not a primary or foreign key
2. SDV's data processor should assign this column to `UniformEncoder`

I think to support this journey, RDT needs to allow `text` then right?

(We should have some separate issues to officially remove `text` from RDT though. It should all be updated to `id`.)

Reply:
Right, I think the question is do we want to include updating SDV to use `'id'` instead of `'text'` when it creates the hyper transformer with this set of changes. That would negate the need for the categorical transformers to support `'text'` (since we will only ever use `'id'`). 

If we remove `'text'` here, then we would need to make the update to the data processor with the changes to SDV to support using the categorical transformers with ID sdtypes. Or, we can leave it as is for now and put up issues in SDV/RDT to align on using `'id'` everywhere.

Reply:
Issue for updating SDV `DataProcessor` is [here](https://github.com/sdv-dev/SDV/issues/2424)